improver_tests/utilities/test_OccurrenceWithinVicinity.py

```diff
@@ -44,187 +44,236 @@
-        )
-        cube.data[0, 0, 2, 1] = 1.0
-        cube.data[1, 1, 1, 3] = 1.0
-        orig_shape = cube.data.copy().shape
-        result = OccurrenceWithinVicinity(self.distance)(cube)
-        self.assertIsInstance(result, Cube)
-        self.assertEqual(result.data.shape, orig_shape)
-        self.assertArrayAlmostEqual(result.data, expected)
-
-    def test_with_multiple_realizations(self):
-        """Test for multiple realizations, so that multiple
-        iterations will be required within the process method."""
-        expected = np.array(
+    )
+    cube.data[0, 1] = 1.0
+    cube.data[2, 3] = 1.0
+    result = OccurrenceWithinVicinity(DISTANCE).maximum_within_vicinity(cube)
+    assert isinstance(result, Cube)
+    assert np.allclose(result.data, expected)
+
+
+def test_fuzzy(cube):
+    """Test for non-binary events to determine where there is an occurrence
+    within the vicinity."""
+    expected = np.array(
+        [
+            [1.0, 1.0, 1.0, 0.0, 0.0],
+            [1.0, 1.0, 1.0, 0.5, 0.5],
+            [0.0, 0.0, 0.5, 0.5, 0.5],
+            [0.0, 0.0, 0.5, 0.5, 0.5],
+            [0.0, 0.0, 0.0, 0.0, 0.0],
+        ]
+    )
+    cube.data[0, 1] = 1.0
+    cube.data[2, 3] = 0.5
+    result = OccurrenceWithinVicinity(DISTANCE).maximum_within_vicinity(cube)
+    assert isinstance(result, Cube)
+    assert np.allclose(result.data, expected)
+
+
+def test_different_distance(cube):
+    """Test for binary events to determine where there is an occurrence
+    within the vicinity for an alternative distance."""
+    expected = np.array(
+        [
+            [1.0, 1.0, 1.0, 1.0, 1.0],
+            [1.0, 1.0, 1.0, 1.0, 1.0],
+            [1.0, 1.0, 1.0, 1.0, 1.0],
+            [0.0, 1.0, 1.0, 1.0, 1.0],
+            [0.0, 1.0, 1.0, 1.0, 1.0],
+        ]
+    )
+    cube.data[0, 1] = 1.0
+    cube.data[2, 3] = 1.0
+    distance = 4000.0
+    result = OccurrenceWithinVicinity(distance).maximum_within_vicinity(cube)
+    assert isinstance(result, Cube)
+    assert np.allclose(result.data, expected)
+
+
+def test_masked_data(cube):
+    """Test masked values are ignored in OccurrenceWithinVicinity."""
+    expected = np.array(
+        [
+            [1.0, 1.0, 1.0, 0.0, 10.0],
+            [1.0, 1.0, 1.0, 1.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0, 1.0],
+            [0.0, 0.0, 0.0, 0.0, 0.0],
+        ]
+    )
+    cube.data[0, 1] = 1.0
+    cube.data[2, 3] = 1.0
+    cube.data[0, 4] = 10.0
+    mask = np.zeros((5, 5))
+    mask[0, 4] = 1
+    cube.data = np.ma.array(cube.data, mask=mask)
+    result = OccurrenceWithinVicinity(DISTANCE).maximum_within_vicinity(cube)
+    assert isinstance(result, Cube)
+    assert isinstance(result.data, np.ma.core.MaskedArray)
+    assert np.allclose(result.data.data, expected)
+    assert np.allclose(result.data.mask, mask)
+
+
+def test_with_land_mask(cube, land_mask_cube):
```

Comment:
As a note for another reviewer, I think that only test_with_land_mask, test_with_land_mask_and_mask, test_with_invalid_land_mask_name and test_with_invalid_land_mask_coords have been added, whilst the other tests have only been refactored.